ipatests/test_webui/test_loginscreen.py

```diff
@@ -0,0 +1,399 @@
+#
+# Copyright (C) 2018  FreeIPA Contributors see COPYING for license
+#
+
+"""
+Test LoginScreen widget and all it's views
+"""
+
+from ipatests.test_webui.ui_driver import UI_driver
+from ipatests.test_webui.ui_driver import screenshot
+import ipatests.test_webui.data_loginscreen as loginscreen
+
+try:
+    from selenium.webdriver.common.by import By
+    from selenium.webdriver.common.keys import Keys
+    from selenium.webdriver.support.wait import WebDriverWait
+except ImportError:
+    pass
+
+import pytest
+from six.moves import urllib
+
+
+@pytest.mark.tier1
+class TestLoginScreen(UI_driver):
+
+    def setup(self, *args, **kwargs):
+        super(TestLoginScreen, self).setup(*args, **kwargs)
+        self.add_test_user()
+
+    def teardown(self, *args, **kwargs):
+        # to log in as administrator
+        if (self.logged_in()):
+            self.logout()
+        else:
+            self.load_url(self.get_base_url())
+        self.delete_test_user()
+        super(TestLoginScreen, self).teardown(*args, **kwargs)
+
+    def delete_test_user(self):
+        """
+        Delete user for tests
+        """
+        self.init_app()
```

Comment:
Calling **self.init_app()** has sense only in the beginning of the test. In case you want to login, just use **self.login()**. Same in the **relogin_with_new_password** method.

Reply:
It was redundant.
Fixed.